applications/dashboard/controllers/Api/RolesApiController.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+use Garden\Schema\Schema;
+use Garden\Web\Data;
+use Garden\Web\Exception\NotFoundException;
+use Garden\Web\Exception\ServerException;
+use Vanilla\Utility\CapitalCaseScheme;
+
+/**
+ * API Controller for the `/roles` resource.
+ */
+class RolesApiController extends AbstractApiController {
+
+    /** @var CapitalCaseScheme */
+    private $caseScheme;
+
+    /** @var Schema */
+    private $idParamSchema;
+
+    /** @var PermissionModel */
+    private $permissionModel;
+
+    /** @var RoleModel */
+    private $roleModel;
+
+    /** @var Schema */
+    private $rolePostSchema;
+
+    /** @var Schema */
+    private $roleSchema;
+
+    /**
+     * RolesApiController constructor.
+     *
+     * @param RoleModel $roleModel
+     */
+    public function __construct(RoleModel $roleModel, PermissionModel $permissionModel) {
+        $this->roleModel = $roleModel;
+        $this->permissionModel = $permissionModel;
+        $this->caseScheme = new CapitalCaseScheme();
+    }
+
+    /**
+     * Delete a role.
+     *
+     * @param int $id The ID of the role.
+     * @return null
+     */
+    public function delete($id) {
+        $this->permission('Garden.Settings.Manage');
+
+        $in = $this->idParamSchema()->setDescription('Delete a role.');
+        $out = $this->schema([], 'out');
+
+        $this->roleByID($id);
+        $this->roleModel->deleteID($id);
+
+        return null;
+    }
+
+    /**
+     * Get a schema instance comprised of all available role fields.
+     *
+     * @return Schema Returns a schema object.
+     */
+    protected function fullSchema() {
+        $schema = Schema::parse([
```

Comment:
I'm amenable to adding the static variable but I'm not sure it'll be helpful.

I think abstracting this is premature since we don't know if it's going to be on every controller or not, yet. I feel like that's something we can refactor at the end if it makes sense rather than trying to guess at it mid-stream, especially when I'm not sure it'll help clarity at this stage.